redwood-protocol-host/src/commonMain/kotlin/app/cash/redwood/protocol/host/ProtocolBridge.kt

```diff
@@ -138,13 +141,6 @@ public class ProtocolBridge<W : Any>(
       }
     }
 
-    // Apply deferred removes. Do this last so properties + modifiers always find their widget,
-    // even if it's marked for death.
-    for (removedId in removedIds) {
-      val removedNode = nodes.remove(removedId)
-      if (removedNode?.reuse == true) pool.addLast(removedNode)
-    }
```

Comment:
This needing to occur a separate step seems like a relic of the change list sorting. The protocol should already guarantee this, as removal from the map would fail any future lookups by ID.